doc/development/guide/architecture.rst

```diff
@@ -0,0 +1,238 @@
+:math:`f(x;\theta)=R^m\rightarrow R^n` that is executed using quantum
+information processing on a quantum device.
+
+Apart from incorporating quantum functions, QNodes also offer custom
+quantum differentiation rules. Using the so-called `parameter-shift rules
+<https://pennylane.ai/qml/glossary/parameter_shift.html>`__, many quantum
+functions can be expressed through the linear combination of other quantum
+functions. As these rules allow quantum gradients to be obtained from
+QNodes, hybrid computations may include QNodes as part of training deep
+learnings models.
+
+PennyLane offers the following qnode types and differentiation rules:
+
+* :class:`~.QubitQNode`: qubit parameter-shift rule
+* :class:`~.CVQNode`: CV parameter-shift rule
+* :class:`~.JacobianQNode`: finite differences
+* :class:`~.DeviceJacobianQNode`: queries the device directly for the gradient
+* :class:`~.PassthruQNode`: classical backpropagation
+* :class:`~.ReversibleQNode`: reversible backpropagation
+
+These QNode types are available to users through the :func:`~.qnode` decorator by
+passing the user-facing ``diff_method`` option. This decorator then uses the
+:func:`~.QNode` constructor function to create the specific type of qnode based on
+the device, interface, and quantum function. If ``diff_method`` option is not
+provided, the QNode constructor function attempts to determine the ``"best"``
+differentiation method, based on the available device and interface.
+
+A widespread representation of quantum circuits is by creating a `Directed
+Acyclic Graph (DAG)
+<https://pennylane.ai/qml/glossary/hybrid_computation.html#directed-acyclic-graphs>`__
+and representing quantum operations within such a graph. Each ``QNode``
+represents the quantum circuit by building such a DAG by creating a
+:class:`~.CircuitGraph` instance.
+
+For further details on QNodes, and a full list of QNodes, refer to the
+:doc:`/code/qml_qnodes` module.
+
+Interfaces
+**********
+
+The integration between classical and quantum computations is encompassed by
+interfaces. QNodes that provide black-box gradient rules are 'wrapped' by an interface function.
+that provide a 'wrapper' around QNodes such. These wrappers further transform
+the ``QNode`` such that the quantum gradient rules of the QNodes are registered
+to the machine learning interface via a custom gradient class or function.
+
+An interface integrates QNodes with external libraries by the following:
+
+* It wraps the QNode, returning a QNode that accepts and returns the core data
+structure of the classical machine learning library (e.g., a TF tensor, Torch
+tensor, Autograd NumPy array, etc).
+* It unwraps the input data structures to simple NumPy arrays, so that the
+quantum device can execute the user's quantum function.
+* It registers the ``QNode.jacobian()`` method as a custom gradient method, so that
+the machine learning library can 'backpropagate' across the QNode, when
+integrated into a classical computation.
```

Comment:
```suggestion
* It wraps the QNode, returning a QNode that accepts and returns the core data
  structure of the classical machine learning library (e.g., a TF tensor, Torch
  tensor, Autograd NumPy array, etc).

* It unwraps the input data structures to simple NumPy arrays, so that the
  quantum device can execute the user's quantum function.

* It registers the ``QNode.jacobian()`` method as a custom gradient method, so that
  the machine learning library can 'backpropagate' across the QNode, when
  integrated into a classical computation.
```